### ISSUE TITLE
[CoE Starter Kit - BUG] App Owner field blank when owner exists - guest users

### ISSUE THREAD

**Issue by bleno123** [microsoft/coe-starter-kit#7457]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

I installed the CoE starter kit a few weeks ago and installation went fine. (Center of Excellence - Core Components version 4.17). I started digging through the associated Power BI Dashboard and for many of my apps, no owner is listed. However when I go into make.power.apps and view properties of that app, an owner exists. Secondly, very few apps show as orphaned but when I once again look at properties of the app, that owner is no longer with my company.

### Expected Behavior

I expect that the app owner would fill in according to what I see via PP admin center and not show as null in Power BI Dashboard. I would also expect to see more orphaned apps.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.17

### What app or flow are you having the issue with?

Not sure which one pulls in app owner or ophaned apps

### What method are you using to get inventory and telemetry?

Cloud flows

AB#2037

**Comment by Jenefer-Monroe** [comment]
Please go to the Admin Command Center > CoE Flows > Inventory. Validate that all flows are turned on and that there are no unmanaged layers as shown here. 
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/598b5024-2e1a-4390-a6b2-49644368c246)

**Comment by bleno123** [comment]
All flows are turned on. There are a few that failed but not sure it would affect the issue I'm having. The failed flows include the following: Admin | Add Maker to Group, HELPER - ObjectOperations, and SetupWizard>ShareApps

**Comment by bleno123** [comment]
There are also no unmanaged layers

**Comment by Jenefer-Monroe** [comment]
Ownership is determined at the time the object is inventoried. Meaning these flows for canvas apps: 
**_Admin | Sync Template v4 (Apps)
SYNC HELPER - Apps_**

Orphan check happens with this flow 
**_CLEANUP - Admin | Sync Template v3 (Orphaned Makers)_**

If someone already left the organization at the time you start the inventory, we are not able to get their information as we use the O365 Users connector and it cannot get users deleted from the AD.

If this still does not answer the question, can you please share some specific cases of concern?

**Comment by bleno123** [comment]
Hi Jenefer. I think that does make sense. I think I was confused because when I reviewed the app or flow detail, it would always list an owner but the dashboard wouldn't reflect that. But you're saying regardless of what the app/flow detail says for an owner, the dashboard is only going to pull in an owner if they were active in Azure AD at the time the object was inventoried, correct?

On a similar note, I've found a couple of examples where no app owner is listed in dashboard but listed in app detail like above. However, the user in this case is listed in AD as a guest user with a different identity (ExternalAzureAD) than a regular employee. I'm assuming that also causes an app owner to show up as blank in the dashboard. Do you know for sure on this one? 

Thanks again for your help. I really appreciate it.

**Comment by Jenefer-Monroe** [comment]
Correct. The product has some extra tables it keeps around and stores old deleted users but we do not access them for our processes. So if they are gone when the object is inventoried they will be blank. However if they leave after you install the kit then we store the users information like Display Name. 

Regarding guest users. That is interesting. I havent tested how they show up I had just assumed they would work. I can take a look.

**Comment by bleno123** [comment]
Regarding maker ID, where do I find that. Is that the same as their object ID in Azure AD? If so it's different.

Here is the output: 
{
  "userisorphan": true,
  "userissystem": false,
  "userisserviceprinciple": false,
  "usercompany": "",
  "userdepartment": "",
  "userdisplayname": "",
  "usercity": "",
  "usercountry": "",
  "userpreferredlanguage": "en-US",
  "userid": "a4869848-d103-437d-9f96-5833e0c0d775",
  "useremail": "",
  "usermanagerid": "",
  "usermanagername": "",
  "userisenabled": true,
  "userupn": "",
  "userstate": ""
}

**Comment by Jenefer-Monroe** [comment]
Yes the GUID expected there is the same as the GUID in the AD. 
Sounds like the issue is that PowerApps isnt returning the correct user to us. 
Please scroll up in the run to this step and look at the output. What is returned for owner?
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/dbfb5ea7-6d8e-41d1-a05d-c43a5726b8e1)

**Comment by bleno123** [comment]
I don't see a "show raw inputs" button but if I look under OUTPUTS, that user name and his info does show up.

**Comment by Jenefer-Monroe** [comment]
What shows up for Owner ID? 
If you browse to the run from flow.microsoft.com instead of powerapps.microsoft.com you will see "show raw outputs"

**Comment by bleno123** [comment]
in my case it's "a4869....." and displays the name of the owner of the app

**Comment by Jenefer-Monroe** [comment]
Sorry and to clarify, a4869... is the guest user's account in AD?

**Comment by Jenefer-Monroe** [comment]
Please go run this flow, put their GUID in the parameter and set Recheck to true
Please open the run and share the path that is followed.
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/f5da585b-9270-44cb-ae47-cf1513b9b297)

**Comment by bleno123** [comment]
> Sorry and to clarify, a4869... is the guest user's account in AD?

I'm trying to figure out what that ID refers to. Their object ID in AD starts with '273'

**Comment by bleno123** [comment]
I ran that last flow. I'm not sure what you mean by "path". Also, happy to jump on the phone too. You may not do that type of troubleshooting but wanted to throw it out there. Thanks.

**Comment by Jenefer-Monroe** [comment]
Unfortunately we are not staffed to have calls with our GitHub users, we are a very small team staffing all 11k installs of the kit. We do best effort to assist but since we arent a product, and rather just a template implementation, we just arent staffed for it. 

That said, no need to run that flow. This was the key, I had misunderstood. 
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/b6c0005a-6fea-4fc9-bed1-b1dfaf814d79)

So for guest users the product does not return the AD GUID of the user but some other GUID. 
So that is the product blocking bug. 

If I were to guess it would be that this is the GUID of the user in the system users table in that envt. 
So if it were in your default envt, for example, you could go to the default envt, open the systemuser table
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/252a9764-7ec6-4580-a3a0-afc651d4920b)

These are probably the two GUIDs we are seeing. 
Can you take a look?

**Comment by bleno123** [comment]
Ok. I did what you asked and in this system users table (I pulled from our default environment, not the environment where CoE is installed), His Azure AD Object ID (according to this table) starts with a486, which is what I mentioned at one point earlier in this thread (see image) even though when I look at his users profile in Azure AD, his object ID starts with 273. I'm not sure why they are different or what that means. This users table also has a column for "isdisabled" and is listed as True. However, his account status in Azure AD is "Enabled". So guess I'm confused as to what all this means.

![image](https://github.com/microsoft/coe-starter-kit/assets/154913304/cd308c18-149b-4bae-ba4d-c0de1af5208a)

**Comment by AmarSaiRam** [comment]
Dear User, 

We noticed that this issue has been pending for quite some time, and we have not received any recent updates from your end. In the absence of further communication, we are proceeding to close this ticket, assuming that your concerns have been addressed or resolved. 
If you continue to experience any difficulties related to this matter, please feel free to open a new support ticket. This will help us ensure prompt and effective assistance, rather than reopening the new Issue. 
Thank you for your understanding and cooperation. 

Best Regards, 
CoE Starter Kit Team
